File: pgm/ABC_Java/ABC_303_A.java
import java.io.*;
// import java.math.*;
// import java.util.*;
// import java.util.Map.*;
import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        final int N = Integer.parseInt (reader.readLine());
        final String[] S = reader.readLine().split ("");
        final String[] T = reader.readLine().split ("");
        final String one = "1";
        final String l = "l";
        final String zero = "0";
        final String o = "o";
        final String Yes = "Yes";
        final String No = "No";
        final int cOne = one.codePointAt (0);
        final int cL = l.codePointAt (0);
        final int cZero = zero.codePointAt (0);
        final int cO = o.codePointAt (0);
        boolean match = IntStream.range (0, N).allMatch (
            i
            -> S[i].equals (T[i]) ||
                   (Math.min (S[i].codePointAt (0), T[i].codePointAt (0)) ==
                        cOne &&
                    Math.max (S[i].codePointAt (0), T[i].codePointAt (0)) ==
                        cL) ||
                   (Math.min (S[i].codePointAt (0), T[i].codePointAt (0)) ==
                        cZero &&
                    Math.max (S[i].codePointAt (0), T[i].codePointAt (0)) ==
                        cO));
        System.out.println (match ? Yes : No);
        return;
    }
}

File: pgm/ABC_R/ABC_303_A.r
con <- file(description = "stdin", open = "r")
arr <- readLines(con = con)
close(con)
N <- arr[1] |>
    as.integer()
S <- strsplit(arr[2], split = "")[[1]]
t <- strsplit(arr[3], split = "")[[1]]
one <- "1"
l <- "l"
zero <- "0"
o <- "o"
Yes <- "Yes"
No <- "No"
c_one <- utf8ToInt(one)
c_l <- utf8ToInt(l)
c_zero <- utf8ToInt(zero)
c_o <- utf8ToInt(o)
match <- TRUE
for (i in seq_len(N)) {
    if (S[i] != t[i]
        && ((min(utf8ToInt(S[i]), utf8ToInt(t[i])) != c_one
                || max(utf8ToInt(S[i]), utf8ToInt(t[i])) != c_l))
        && ((min(utf8ToInt(S[i]), utf8ToInt(t[i])) != c_zero
                || max(utf8ToInt(S[i]), utf8ToInt(t[i])) != c_o))) {
        match <- FALSE
        break
    }
}
if (match) {
    Yes |>
        cat("\n", sep = "")
} else {
    No |>
        cat("\n", sep = "")
}
q("no")